File: frontend/lib/model/job.model.dart
class Job {
  final String id;
  final String role;
  final String company;
  final String location;
  final String locationType;
  final String salary;
  final String jobType;
  final String jobDescription;
  final String qualifications;
  final String requiredSkills;
  final String responsibilities;
  final int vacancies;
  final String experience;
  final DateTime deadline;
  final DateTime posted;
  final String logoUrl;

  Job({
    required this.id,
    required this.role,
    required this.company,
    required this.location,
    required this.locationType,
    required this.salary,
    required this.jobType,
    required this.jobDescription,
    required this.qualifications,
    required this.requiredSkills,
    required this.responsibilities,
    required this.vacancies,
    required this.experience,
    required this.deadline,
    required this.posted,
    required this.logoUrl
  });

  factory Job.fromJson(Map<String, dynamic> json) {
    return Job(
      id: json['_id'],
      role: json['role'],
      company: json['company'],
      location: json['location'],
      locationType: json['location_type'],
      salary: json['salary'],
      jobType: json['job_type'],
      jobDescription: json['job_description'],
      qualifications: json['qualifications'],
      requiredSkills: json['required_skills'],
      responsibilities: json['responsibilities'],
      vacancies: json['vacancies'],
      experience: json['experience'],
      deadline: DateTime.parse(json['deadline']),
      posted: DateTime.parse(json['posted']),
      logoUrl: json['logo']
    );
  }
}

File: backend/model/jobModel.js
const mongoose = require('mongoose');
const db = require('../config/database');

const jobSchema = new mongoose.Schema({
  role: {
    type: String,
    required: true
  },
  company: {
    type: String,
    required: true
  },
  location: {
    type: String,
    required: true
  },
  location_type: {
    type: String,
    enum: ['remote', 'in-office', 'hybrid'],
    required: true
  },
  salary: {
    type: String,
    required: true
  },
  job_type: {
    type: String,
    enum: ['Full-time', 'Part-time'],
    required: true
  },
  job_description: {
    type: String,
    required: true
  },
  qualifications: {
    type: String,
    required: true
  },
  required_skills: {
    type: String,
    required: true
  },
  responsibilities: {
    type: String,
    required: true
  },
  vacancies: {
    type: Number,
    required: true
  },
  experience: {
    type: String,
    required: true
  },
  deadline: {
    type: Date,
    required: true,
    index: { expireAfterSeconds: 0 } 
  },
  posted: {
    type: Date,
    default: Date.now(),
  },
  postedAgo: {
    type: Number
  },
  logo: {
    type: String
  }
});

jobSchema.pre('save', function(next) {
  if (!this.postedAgo) {
    const now = Date.now();
    const postedTime = this.posted.getTime();
    const diffInMilliseconds = now - postedTime;
    const diffInDays = Math.floor(diffInMilliseconds / (1000 * 60 * 60 * 24)); // Convert milliseconds to days
    this.postedAgo = diffInDays;
  }
  next();
});

const job = db.model('Job',jobSchema);

module.exports = job;
